### PR TITLE
2 hook up form fields

### LINKED ISSUE #2: Hook up form fields
The form at /new/ needs to have the First Name, Last Name and Upload fields hooked up. I've hard-coded them here so I could style them. Originally it was just email address. Those fields should be output into the database and admin area so they can be exported after the promo is over. (the default admin login creds are in the readme and should probably be changed.)

The upload field should accept pdf and image files only I guess (png, gif, jpg, etc)

### PR DESCRIPTION
**Pull request by RanSolo**
fixes #2 
@capitaltechnologyservices/ruby-developers ready for review.

screenshot 2016-02-19 12 39 03
screenshot 2016-02-19 12 39 11

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -13,54 +13,46 @@ def new
...
-            else
-                format.html { redirect_to root_path, :alert => "Something went wrong!" }
-            end
+        if cur_ip.count > 2
+          return redirect_to root_path
+        else
+          cur_ip.count = cur_ip.count + 1
+          cur_ip.save
+        end
+        @user = User.create(params[:user])
```
Comment (RanSolo) [inline app/controllers/users_controller.rb]: line of interest

File: app/views/users/new.html.erb [hunk 2]
```diff
@@ -59,34 +58,30 @@
 		        <p class="go-to brandon">In the meantime, head to  <a href="http://www.themindfulnessedge.com/">www.matttenney.com</a></p>
 		        <p class="again brandon">Thank You Again!</p>
 		    <% else %>
-		        <%= form_for @user, :url => { :action => "create" } do |f| %>
-					<label for="first_name">First Name</label>
-		            <input class="first_name" id="user_first_name" name="user[first_name]" placeholder="First Name" size="30" type="text">
+		      <%= form_for @user, :url => { :action => "create" },  :html => { :multipart => true } do |f| %>
```
Comment (RanSolo) [inline app/views/users/new.html.erb]: Here is the form_for.

File: app/views/users/refer.html.erb [hunk 3]
```diff
@@ -1,9 +1,10 @@
 <div class="container">
 	<div class="clearfix upper_page">
-		<h1 class="page_title">Thanks for buying the book!</h1>
+		<h1 class="page_title"><%= @user.first_name %>, thanks for buying the book!</h1>
+		<%= image_tag @user.receipt.url(:thumb) %>
```
Comment (RanSolo) [inline app/views/users/refer.html.erb]: Here is the image_tag.

File: app/views/users/new.html.erb [hunk 4]
```diff
@@ -59,34 +58,30 @@
...
-					</div><!-- form_section -->
-
-		            <input class="submit signup_btn" name="commit" type="submit" value="Sign Up Now" />
-					
-		        <% end %>
-		    <% end %>
+						<div class="form_section clearfix">
+							<h3>Upload Your Receipt</h3>
+							<p>Please upload a PDF, photo, or screenshot of your receipt here.</p>
+							<%= f.file_field :reciept, :name => "fileToUpload", :id => 'fileToUpload', :class => "clear" %>
```
Comment (RanSolo) [inline app/views/users/new.html.erb]: Here is the file upload

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (RanSolo) [inline app/controllers/users_controller.rb]: line 34 has been changed from @user = User.new(:email => params[:user][:email])

**Comment by drewmoore** [comment]
:+1: Adding s3.

**Comment by jchapin** [comment]
Merge conflicts.

**Comment by jchapin** [comment]
Merge conflicts worked out, changes to the Gemfile, also changes to the production environment config and the devise initializer needed to be de-duplicated. :+1: PR sets up AWS SDK and paperclip and also changes params in the controller for certain form fields setup on the front page.

**Pull request merged**